provider/internal/image_test.go

```diff
@@ -273,6 +273,48 @@ func TestImageLifecycle(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "context defaults to current directory (pulumi-docker-build#78)",
+			client: func(t *testing.T) Client {
+				ctrl := gomock.NewController(t)
```

Comment:
Side question, recommend using this generally at Pulumi? https://github.com/uber-go/mock 

I'm not a big fan of mocking prior to working in Go but in some situations it is practical, and I'd love a library recommendation.

Reply:
Definitely! It can be a life saver for larger interfaces. I initially introduced it in the service and it's [caught on](https://github.com/search?q=org%3Apulumi+%22go.uber.org%2Fmock%22&type=code) since then.